src/lib/bootloader/udev_mapping.rb

```diff
@@ -34,17 +34,28 @@ def to_hash
     end
 
     # Converts full udev name to kernel device ( disk or partition )
-    # @param dev [String] device udev or kernel one like /dev/disks/by-id/blabla
+    # @param dev [String] device udev or kernel one like /dev/disk/by-id/blabla
     # @raise when device have udev format but do not exists
     def to_kernel_device(dev)
-      return dev if dev !~ /^\/dev\/disk\/by-/
-
-      all_devices[dev] or raise "Unknown udev device #{dev}"
+      # for non-udev devices try to see specific raid names (bnc#944041)
+      if dev =~ /^\/dev\/disk\/by-/
+        all_devices[dev] or raise "Unknown udev device #{dev}"
+      else
```

Comment:
idea of whole this class is to have quick cache for udev mapped devices. It is relict from past that speed up this translation for system with incredible huge amount of disks. So any call outside just slow down its performance.